packages/apollo-link-batch/src/batchLink.ts

```diff
@@ -58,6 +58,10 @@ export class BatchLink extends ApolloLink {
       batchMax: this.batchMax,
       batchHandler: fetchParams.batchHandler,
     });
+
+    if (fetchParams.batchHandler.length <= 1) {
+      this.request = operation => this.batcher.enqueueRequest({ operation });
```

Comment:
Is overriding `request` in the constructor a common thing? Instead of replacing `request` here, would it be better to check `fetchParams.batchHandler.length <= 1` inside the `public request` method?

Reply:
I'm a bit confused as to why we need to replace the `request` function here at all - what is the `batchHandler.length` measuring? Like the number of arguments to `batchHandler`?

Reply:
Yep it's a count of how many arguments the function has. It's how we determine if a link is terminating or non-terminating(based on if request accepts a forward or not).

Admittedly it is a pretty poor way of doing termination/non-termination in this case. If there are better ways to determine if a link is terminating or modifying the public methods signature other than overriding, please share!